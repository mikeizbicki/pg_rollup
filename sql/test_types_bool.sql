SET client_min_messages TO WARNING;
create or replace language plpython3u;
create extension if not exists pg_rollup;

create table testbool (
    id serial primary key,
    a bool,
    b bool,
    c bool,
    d bool
);

create materialized view testbool_rollup1 as (
    select 
        bool_and(a) as and_a,
        bool_and(b) as and_b,
        bool_and(c) as and_c,
        bool_and(d) as and_d,
        bool_or(a) as or_a,
        bool_or(b) as or_b,
        bool_or(c) as or_c,
        bool_or(d) as or_d
    from testbool
);

insert into testbool (a,b,c,d) values (TRUE,FALSE,NULL,NULL);
select assert_rollup('testbool_rollup1');
insert into testbool (a,b,c,d) values (TRUE,FALSE,NULL,NULL);
select assert_rollup('testbool_rollup1');
insert into testbool (a,b,c,d) values (FALSE,TRUE,TRUE,FALSE);
select assert_rollup('testbool_rollup1');
insert into testbool (a,b,c,d) values (NULL,NULL,FALSE,TRUE);
select assert_rollup('testbool_rollup1');
insert into testbool (a,b,c,d) values (TRUE,FALSE,NULL,NULL);
select assert_rollup('testbool_rollup1');

drop table testbool cascade;
