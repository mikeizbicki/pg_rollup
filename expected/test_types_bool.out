create or replace language plpython3u;
create extension if not exists pg_rollup;
NOTICE:  extension "pg_rollup" already exists, skipping
create temporary table testbool (
    id serial primary key,
    a bool,
    b bool,
    c bool,
    d bool
);
select create_rollup(
    'testbool',
    'testbool_rollup1',
    rollups => $$
        bool_and(a),
        bool_and(b),
        bool_and(c),
        bool_and(d),
        bool_or(a),
        bool_or(b),
        bool_or(c),
        bool_or(d)
    $$
);
NOTICE:  view "testbool_rollup1_groundtruth_raw" will be a temporary view
NOTICE:  view "testbool_rollup1" will be a temporary view
NOTICE:  view "testbool_rollup1_groundtruth" will be a temporary view
 create_rollup 
---------------
 
(1 row)

insert into testbool (a,b,c,d) values (TRUE,FALSE,NULL,NULL);
select assert_rollup('testbool_rollup1');
 assert_rollup 
---------------
 
(1 row)

insert into testbool (a,b,c,d) values (TRUE,FALSE,NULL,NULL);
select assert_rollup('testbool_rollup1');
 assert_rollup 
---------------
 
(1 row)

insert into testbool (a,b,c,d) values (FALSE,TRUE,TRUE,FALSE);
select assert_rollup('testbool_rollup1');
 assert_rollup 
---------------
 
(1 row)

insert into testbool (a,b,c,d) values (NULL,NULL,FALSE,TRUE);
select assert_rollup('testbool_rollup1');
 assert_rollup 
---------------
 
(1 row)

insert into testbool (a,b,c,d) values (TRUE,FALSE,NULL,NULL);
select assert_rollup('testbool_rollup1');
 assert_rollup 
---------------
 
(1 row)

